File: MediCareWebSpa/src/app/DTOs/models/doctor.ts
import { Appointment } from "./appointment";
import { DoctorsAvailability } from "./doctors-availability";
import { Speciality } from "./speciality";
import { User } from "./user";

export class Doctor {
  userId: number;
  user: User;
  specialityId: number;
  speciality: Speciality;
  employmentDate: Date;
  appointments: Appointment[] = [];
  doctorsAvailabilities: DoctorsAvailability[] = [];

  constructor(
    userId: number,
    user: User,
    specialityId: number,
    speciality: Speciality,
    employmentDate: Date
  ) {
    this.userId = userId
    this.user = user;
    this.specialityId = specialityId;
    this.speciality = speciality;
    this.employmentDate = employmentDate;
  }
}

File: MediCareWebApi/Models/Doctor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class Doctor
	{
		[Key]
		public int UserId { get; set; }

		[ForeignKey("UserId")]
		public User User { get; set; }

		[Required]
		public DateTime EmploymentDate { get; set; }

		[Required]
		public int SpecialityId { get; set; }

		[ForeignKey("SpecialityId")]
		public Speciality Speciality { get; set; }

		public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

		public ICollection<DoctorsAvailability> DoctorsAvailabilities { get; set; } = new List<DoctorsAvailability>();
	}
}
